nixos/modules/services/misc/klipper.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.klipper;
+  package = pkgs.klipper;
+in
+{
+  ##### interface
+  options = {
+    services.klipper = {
+      enable = mkEnableOption "Klipper, the 3D printer firmware";
+
+      user = mkOption {
+        type = types.str;
+        default = "klipper";
+        description = "User for the daemon.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "klipper";
+        description = "Group for the daemon.";
+      };
+
+      logDir = mkOption {
+        type = types.path;
+        default = "/var/log/klipper";
+        description = "Log directory of the daemon.";
+      };
+
+      extraConfig = mkOption {
+        type = types.attrs;
+        default = {};
+        description = "Extra options which are added to klipper's configuration file.";
+      };
+    };
+  };
+
+  ##### implementation
+  config = mkIf cfg.enable {
+    environment.etc."klipper.cfg".text = lib.toINI { mkKeyValue = ":"; } cfg.extraConfig;
+
+    users.users = optionalAttrs (cfg.user == "klipper") {
```

Comment:
Why does this create the octoprint user and group iff the service's user and group are set to klipper?

If there's no reason, maybe `users.users.octoprint = { group = cfg.group; uid = config.ids.uids.${cfg.user}; };` would work?

Reply:
It was an accident since I used octoprint's service as a start point.

Klipper should run as root, and so I removed this potion of the config altogether.